src/components/ReservationPage.tsx

```diff
@@ -0,0 +1,386 @@
+  const [currentProject, setCurrentProject] = useState(initialProject);
+
+  useEffect(() => {
+    fetch(`/api/project_groups/${currentProject?.id}`)
+      .then((response) => response.json())
+      .then((groups) => {
+        const usergroups = groups.map(
+          (group: UserGroup) => new UserGroup(group)
+        );
+        setGroups(usergroups);
+      })
+      .catch(console.error);
+  }, [currentProject, user]);
+
+  const classes = useStyles();
+
+  const [liveToggle, setLiveValue] = React.useState("yes");
+  const [guestToggle, setGuestValue] = React.useState("no");
+  const [notesToggle, setNotesValue] = React.useState("no");
+
+  const changeProject = (id: number): void => {
+    const proj = state.projects.filter(function (project) {
+      // eslint-disable-next-line
+      return project.id == id;
+    });
+    setCurrentProject(proj[0]);
+  };
+
+  const liveRoomChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setLiveValue((event.target as HTMLInputElement).value);
+  };
+
+  const showGuests = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setGuestValue((event.target as HTMLInputElement).value);
+    const val = (event.target as HTMLInputElement).value;
+    const element = document.getElementById("guestInput");
+    if (element != null) {
+      if (val === "yes") {
+        element.style.display = "block";
+      } else {
+        element.style.display = "none";
+      }
+    }
+  };
+
+  const showNotes = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setNotesValue((event.target as HTMLInputElement).value);
+    const val = (event.target as HTMLInputElement).value;
+    const element = document.getElementById("notesInput");
+    if (element != null) {
+      if (val === "yes") {
+        element.style.display = "block";
+      } else {
+        element.style.display = "none";
+      }
+    }
+  };
+
+  // Once we have projects, if we still don't have a default currentProject, assign it now.
+  if (projects[0]) {
+    if (currentProject === undefined) {
+      setCurrentProject(projects[0]);
+    }
+  }
+  const phoneRegExp = /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;
+
+  return (
+    <Dialog
+      fullScreen
+      open={state.reservationPageIsOpen}
+      TransitionComponent={transition}
+    >
+      <Toolbar>
+        <IconButton
+          edge="start"
+          color="inherit"
+          aria-label="close1"
+          onClick={(): void =>
+            dispatch({ type: CalendarAction.CloseReservationPage })
+          }
+        >
+          <CloseIcon />
+        </IconButton>
+        <Typography variant="h6">Make Reservation</Typography>
+      </Toolbar>
+      <React.Fragment>
+        <DialogContent>
+          <Formik
+            initialValues={{ phone: "", description: "", guests: "" }}
+            onSubmit={(values, { setSubmitting }) => {
+              return undefined;
```

Comment:
clearly a work in progress, but just noting that `undefined` does not need to be explicitly declared here
```suggestion
              return;
```

Reply:
this has been changed to a simulated server response and now changes the modal to a reservation confirmation on "Server success" (which is 100% currently)